### ISSUE TITLE
{Linked:Bug9715653} Talkback does not announce the name and role of the year button.

### ISSUE THREAD

**Issue by vaishali1397** [microsoft/AdaptiveCards#8007]
### Target Platforms

Android

### SDK Version

2022.10.31.1 (58691996)

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
OS: Android
Application: Adaptive Cards
Screen Reader: Talkback

**Repro steps:**

1. Open Adaptive card Android Visualizer.
2. Navigate to the 'Import' card and activate it.
3. Select the provided JSON.
4. Navigate to the 'Visual tab item'.
5. Observe the issue.

**Actual Result:**
Talkback narrates the year button as the selected year.

**Expected Result:**
Talkback should announce the name and role of the year button.

**User Impact:**
Screen reader user will get impacted as user will not be able to understand the purpose and its relationship with the content easily.

**MAS Reference link:**
https://liquid.microsoft.com/Web/Object/Read/MS.Accessibility/Requirements/04.01.02

"Have feedback to share on Bugs? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0) “.

### Screenshots

https://user-images.githubusercontent.com/104483005/199501062-bc31d7d9-6fd7-492b-81a3-397552d98475.mp4

### Card JSON

```JSON
{
  "schema": "http=//adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.2",
  "body": [
    {
      "id": "",
      "isVisible": true,
      "separator": false,
      "spacing": "default",
      "type": "TextBlock",
      "textStyle": "default",
      "wrap": true,
      "text": "Write a personal note to put your request in context",
      "size": "medium",
      "weight": "bolder"
    },
    {
      "id": "",
      "isVisible": true,
      "separator": false,
      "spacing": "default",
      "type": "Container",
      "items": [
        {
          "id": "",
          "isVisible": true,
          "separator": false,
          "spacing": "large",
          "type": "Container",
          "items": [
            {
              "id": "",
              "isVisible": true,
              "separator": false,
              "spacing": "default",
              "type": "Container",
              "items": [
                {
                  "id": "",
                  "isVisible": true,
                  "separator": false,
                  "spacing": "Medium",
                  "type": "ColumnSet",
                  "bleed": false,
                  "columns": [
                    {
                      "type": "Column",
                      "separator": false,
                      "spacing": "default",
                      "verticalContentAlignment": "center",
                      "width": "60px",
                      "isVisible": true,
                      "items": [
                        {
                          "id": "",
                          "isVisible": true,
                          "separator": false,
                          "spacing": "default",
                          "width": "50px",
                          "type": "Image",
                          "height": "auto",
                          "horizontalAlignment": "center",
                          "size": "auto",
                          "style": "person",
                          "url": "https://localhost:4321/node_modules/myhub-viva-library/dist/GP_c9e7cb188bc597fb86474b473854e822.png",
                          "altText": "Employee Image of Geok-Kheng Selva Kumar P"
                        }
                      ]
                    },
                    {
                      "type": "Column",
                      "separator": false,
                      "spacing": "default",
                      "verticalContentAlignment": "center",
                      "isVisible": true,
                      "items": [
                        {
                          "id": "",
                          "isVisible": true,
                          "separator": false,
                          "spacing": "default",
                          "type": "TextBlock",
                          "textStyle": "default",
                          "wrap": true,
                          "text": "Geok-Kheng Selva Kumar P",
                          "weight": "bolder",
                          "size": "medium",
                          "color": "default"
                        },
                        {
                          "id": "",
                          "isVisible": true,
                          "separator": false,
                          "spacing": "none",
                          "type": "TextBlock",
                          "textStyle": "default",
                          "wrap": true,
                          "text": "SENIOR PROGRAM MANAGER",
                          "weight": "lighter",
                          "size": "small"
                        }
                      ]
                    }
                  ]
                }
              ],
              "verticalContentAlignment": "center",
              "bleed": false
            },
            {
              "id": "ExpirationDate_346409",
              "isVisible": true,
              "separator": false,
              "spacing": "Medium",
              "type": "Input.Date",
              "isRequired": false,
              "value": "2022-11-27",
              "min": "2022-11-02",
              "label": "Expiration date"
            },
            {
              "id": "Notes_346409",
              "isVisible": true,
              "separator": false,
              "spacing": "Medium",
              "type": "Input.Text",
              "isRequired": false,
              "isMultiline": true,
              "style": "text",
              "value": "",
              "placeholder": "Personalize your request with a note ...",
              "maxLength": 500
            }
          ],
          "verticalContentAlignment": "center",
          "bleed": false
        }
      ],
      "verticalContentAlignment": "center",
      "bleed": false
    },
    {
      "id": "",
      "isVisible": true,
      "separator": false,
      "spacing": "large",
      "type": "ActionSet",
      "actions": [
        {
          "style": "positive",
          "id": "nextButtonID",
          "title": "Next",
          "altText": "Next",
          "type": "Action.Submit",
          "associatedInputs": "auto",
          "data": {
            "id": "nextButtonID",
            "name": "Request Feedback Next Button Clicked"
          }
        }
      ]
    }
  ],
  "actions": [],
  "refresh": null
}
```

**Comment by anna-dingler** [comment]
Hi @vaishali1397,

Our `Input.Date` control uses the default `DatePicker` control from android. We do not have any control over the content of the `DatePicker` and cannot update the button you are referring to.

I also confirmed that a similar behavior (name and role are not announced) can be observed in the settings application.

https://user-images.githubusercontent.com/98650930/228013545-6496f23e-4d5d-44bb-821e-9a8b4f8e1027.mp4

I am closing this issue for now. Please let me know if you have any questions - thanks!

**Comment by vagpt** [comment]
As per the above comment, adding A11yTTValidated tag.
